src/shogun/io/ARFFFile.h

```diff
@@ -0,0 +1,544 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef SHOGUN_ARFFFILE_H
+#define SHOGUN_ARFFFILE_H
+
+#include <shogun/base/init.h>
+#include <shogun/base/variant.h>
+#include <shogun/features/CombinedFeatures.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/lib/SGVector.h>
+
+#include <cctype>
+#include <fstream>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace shogun
+{
+	/** Contains miscellaneous string manipulation functions using the STL and
+	 * Java to C++ date format utilities */
+	namespace arff_detail
+	{
+		/**
+		 * Checks if string is blank
+		 * @param line to check
+		 * @return bool whether line is empty
+		 */
+		SG_FORCED_INLINE bool string_is_blank(const std::string& line)
+		{
+			return line.find_first_not_of(" \t\r\f\v") == std::string::npos;
+		}
+
+		/**
+		 * Checks if the character in the lhs is in the rhs
+		 * @param lhs the single character to find
+		 * @param rhs a string with the characters to compare against the lhs
+		 * @return whether the character of the lhs is in the rhs
+		 */
+		SG_FORCED_INLINE bool char_in_string(char lhs, const std::string& rhs)
```

Comment:
minor: maybe `contains`? since the types are already in the signature...

Reply:
I agree!